File: assignment02/p1.js
// https://www.hackerrank.com/challenges/counting-valleys/problem

const countingValleys = function (steps, path){
    let level = 0, valleys = 0;

    for(let i=0; i<steps; i++){
        if(path[i] == 'U')
            level++;
        else
            level--;

        if(level==0 && path[i] == 'U')
            valleys += 1;
    }

    return valleys;
}

// Test Case 1
console.log(countingValleys(8,'UDDDUDUU'));

// Test Case 2
console.log(countingValleys(8,'DDDDUUUU'));

File: assignment04/p1.cpp
// https://www.hackerrank.com/challenges/staircase/problem

#include<bits/stdc++.h>

void staircase(int n)
{
    for (int i = 0; i < n; i++)
    {
        for (int j = i; j < n-1; j++)
            printf(" ");
        for (int k = 0; k < i + 1; k++)
            printf("#");
        printf("\n");
    }
}

int main(){
    staircase(6);
    
    return 0;
}